lib/types/sign_command.ex

```diff
@@ -15,5 +20,25 @@ defmodule Kadena.Types.SignCommand do
 
   @impl true
   def new(%SignatureWithHash{} = signature), do: %__MODULE__{command: signature}
-  def new(_sig), do: {:error, :invalid_sign_command}
+
+  def new(args) when is_list(args) do
+    hash = Keyword.get(args, :hash, nil)
+    sig = Keyword.get(args, :sig, nil)
+    pub_key = Keyword.get(args, :pub_key, nil)
+
+    case validate(hash, sig, pub_key) do
+      {:error, _error} -> {:error, :invalid_sign_command}
+      signature -> %__MODULE__{command: SignatureWithHash.new(signature)}
+    end
+  end
+
+  def new(_args), do: {:error, :invalid_sign_command}
+
+  @spec validate(hash :: String.t(), sig :: Signature.t(), pub_key :: String.t()) ::
```

Comment:
@santiagoboterokommit IMO, **validate** is not an appropriate name for this function. What about `set_signature_type` ??

Reply:
@juanhurtado10 I think it looks better, thank you.